ts/src/pro/gate.ts

```diff
@@ -767,6 +772,145 @@ export default class gate extends gateRest {
         client.resolve (this.balance, messageHash);
     }
 
+    async watchPositions (symbols: string[] = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name gate#watchPositions
+         * @see https://www.gate.io/docs/developers/futures/ws/en/#positions-subscription
+         * @see https://www.gate.io/docs/developers/delivery/ws/en/#positions-subscription
+         * @see https://www.gate.io/docs/developers/options/ws/en/#positions-channel
+         * @description watch all open positions
+         * @param {[string]|undefined} symbols list of unified market symbols
+         * @param {object} params extra parameters specific to the gate api endpoint
+         * @returns {[object]} a list of [position structure]{@link https://docs.ccxt.com/en/latest/manual.html#position-structure}
+         */
+        await this.loadMarkets ();
+        let market = undefined;
+        symbols = this.marketSymbols (symbols);
+        const payload = [ '!' + 'all' ];
+        if (!this.isEmpty (symbols)) {
+            market = this.getMarketFromSymbols (symbols);
+        }
+        let type = undefined;
+        let query = undefined;
+        [ type, query ] = this.handleMarketTypeAndParams ('watchPositions', market, params);
+        const typeId = this.getSupportedMapping (type, {
+            'future': 'futures',
+            'swap': 'futures',
+            'option': 'options',
+        });
+        let messageHash = type + ':positions';
```

Comment:
@pcriadoperez in this case if no symbol is provided we should default to swap since this is a swap-only method (when the defaultType is spot)